src/frontend/Parser.y

```diff
@@ -1045,6 +1048,20 @@ checkAsciiOf e tk = do
       RWS.tell [Error ("ascii_of requires argument to be a miracle or a sign") (T.position tk)]
       return T.TypeError
 
+checkIsActive :: G.Expr -> T.Token -> ST.ParserMonad T.Type
+checkIsActive e tk = do
+  t <- getType e
+  case t of
+    T.TypeError -> return T.TypeError
+    _ ->
+      if isUnionAttr e then return T.TrileanT
+      else do
+        RWS.tell [Error ("is_active requires argument to be a union attribute") (T.position tk)]
+        return T.TypeError
+  where isUnionAttr (G.Expr {G.expAst=(G.Access r _)}) = isUnion r
+        isUnionAttr _ = False
+        isUnion (G.Expr {G.expType=(T.UnionT _)}) = True
+        isUnion _ = False
```

Comment:
These functions don't need to have a parameter for `e`: it is in their scope already.